pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -223,6 +224,10 @@ public void setMinStreamAvailableCheckDelay(int delay) {
 
   private Socket createSocket(int timeout) throws IOException {
     Socket socket = socketFactory.createSocket();
+    String localSocketAddress = hostSpec.getLocalSocketAddress();
+    if (localSocketAddress != null) {
```

Comment:
Should this also check if the string is not-empty? That way an explicit property value of empty string would be skipped. Otherwise it will attempt to use the empty string value.